internal/services/apimanagement/api_management_workspace_group_resource.go

```diff
@@ -0,0 +1,275 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package apimanagement
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apimanagement/2024-05-01/group"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/schemaz"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type ApiManagementWorkspaceGroupModel struct {
+	Name                     string `tfschema:"name"`
+	ApiManagementWorkspaceId string `tfschema:"api_management_workspace_id"`
+	DisplayName              string `tfschema:"display_name"`
+	Description              string `tfschema:"description"`
+	ExternalId               string `tfschema:"external_id"`
+	Type                     string `tfschema:"type"`
+}
+
+type ApiManagementWorkspaceGroupResource struct{}
+
+var _ sdk.ResourceWithUpdate = ApiManagementWorkspaceGroupResource{}
+
+var _ sdk.ResourceWithCustomizeDiff = ApiManagementWorkspaceGroupResource{}
+
+func (r ApiManagementWorkspaceGroupResource) ResourceType() string {
+	return "azurerm_api_management_workspace_group"
+}
+
+func (r ApiManagementWorkspaceGroupResource) ModelObject() interface{} {
+	return &ApiManagementWorkspaceGroupModel{}
+}
+
+func (r ApiManagementWorkspaceGroupResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return group.ValidateWorkspaceGroupID
+}
+
+func (r ApiManagementWorkspaceGroupResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": schemaz.SchemaApiManagementChildName(),
+
+		"api_management_workspace_id": commonschema.ResourceIDReferenceRequiredForceNew(&group.WorkspaceId{}),
+
+		"display_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"external_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"description": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"type": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ForceNew: true,
+			Default:  string(group.GroupTypeCustom),
+			// `group.GroupTypeSystem` is excluded here because system groups are predefined by API Management
+			// and cannot be created through the API.
```

Comment:
is it possible to import a `GroupTypeSystem` and modify its description? I assume no but please do a test to confirm, thanks